packages/filesystem/src/browser/file-tree/file-tree-model.ts

```diff
@@ -109,4 +111,51 @@ export class FileTreeModel extends TreeModel implements LocationService {
         return true;
     }
 
+    upload(destination: string, file: File, fileName: string) {
+        const fileToCreate = destination + "/" + fileName;
+        this.fileSystem.exists(fileToCreate).then(exist => {
+            this.fileSystem.createFile(fileToCreate).then(() => {
+
+                // Saving contents
+                this.fileSystem.getFileStat(fileToCreate).then(currentStat => {
+                    const reader = new FileReader();
+
+                    reader.onload = e => {
+                        const fileContent: ArrayBuffer = reader.result;
+                        const buffer64 = base64.fromByteArray(new Uint8Array(fileContent));
+                        const options = { encoding: 'base64' };
+                        this.fileSystem.setContent(currentStat, buffer64, options).then(() => Promise.resolve());
+                    };
+                    reader.readAsArrayBuffer(file);
+                });
+            },
+                (error: Error) => {
+                    // Error: Error occurred while creating the file. File already exists
+                    // Ask the end-user to replace the file or not
+                    this.shouldOverwrite(fileToCreate).then(replace => {
+                        if (replace) {
+                            // Rename the existing file with a new extension.
+                            const mvFile = fileToCreate + ".orig";
+                            const overwrite = { overwrite: true };
+                            this.fileSystem.move(fileToCreate, mvFile, overwrite);
```

Comment:
the user got asked about overwriting the file, but now the previous one is still there with a `.orig` suffix, no?

Reply:
It could occurs when dragging a file to a new location, there was an existing file with the same name. I keep a version of the original file with the extension ".orig". I think it is appropriate and it allows the user to compare the file in case he made a mistake.

Reply:
No more move to "file.orig", no question asked either.